api/kv_v2.go

```diff
@@ -427,9 +512,9 @@ func extractCustomMetadata(secret *Secret) (map[string]interface{}, error) {
 		if !ok {
 			return nil, fmt.Errorf("unexpected type for 'metadata' element: %T (%#v)", mI, mI)
 		}
-		cmI, ok = mM["custom_metadata"]
+		cmI, ok = mM[customMetadataKey]
 		if !ok {
-			return nil, fmt.Errorf("metadata missing expected field \"custom_metadata\":%v", mM)
+			return nil, fmt.Errorf("metadata missing expected field \"%s\":%v", customMetadataKey, mM)
```

Comment:
```suggestion
			return nil, fmt.Errorf("metadata missing expected field %q: %v", customMetadataKey, mM)
```

Reply:
Ooh, I didn't know about this one, thanks!